recipe/meta.yaml

```diff
@@ -7,48 +12,48 @@ package:
 source:
   fn: mpfr-{{ version }}.tar.bz2
   url: http://ftp.gnu.org/gnu/mpfr/mpfr-{{ version }}.tar.gz
-  sha256: ae26cace63a498f07047a784cd3b0e4d010b44d2b193bab82af693de57a19a78
-  patches:
-  - patches/build-vc14.patch
-  - patches/corei5.patch
+  sha256: 116715552bd966c85b417c424db1bbdf639f53836eb361549d1f8d6ded5cb4c6
 
 build:
-  number: 1
-  skip: True  # [win and vc<14]
+  number: 3
   run_exports:
     - {{ pin_subpackage("mpfr") }}
+  missing_dso_whitelist:
+    - $RPATH/ld64.so.1  # [s390x]
 
 requirements:
   build:
     - libtool         # [unix]
     - m4              # [unix]
-    - make            # [unix]
-    - {{ compiler('c') }}
-    - patch       # [not win]
-    - m2-patch    # [win]
+    - gnuconfig
+    - {{ posix }}make
+    - {{ posix }}base             # [build_platform.startswith("win-")]
+    - {{ posix }}diffutils        # [build_platform.startswith("win-")]
+    - {{ compiler('c') }}         # [unix]
+    - {{ compiler('ucrt64_c') }}   # [win]
   host:
-    - gmp             # [unix]
-    - mpir            # [win]
+    - {{ native }}gmp
 
 test:
   requires:
-    - {{ compiler('c') }}
-    - libtool         # [unix]
-    - m4              # [unix]
-
+    - {{ compiler('c') }}         # [unix]
+    - {{ compiler('ucrt64_c') }}   # [win]
   commands:
     - if not exist %LIBRARY_INC%\\mpfr.h exit 1                # [win]
     - if not exist %LIBRARY_LIB%\\mpfr.lib exit 1              # [win]
-    - if not exist %LIBRARY_BIN%\\mpfr.dll exit 1              # [win]
-    - if not exist %LIBRARY_LIB%\\mpfr_static.lib exit 1       # [win]
-    - test -f ${PREFIX}/lib/libmpfr.a                          # [unix]
+    - if not exist %LIBRARY_BIN%\\libmpfr-6.dll exit 1         # [win]
```

Comment:
I believe that `configure` generates the `libFOO-N.dll` name which is different to the naive MS `libFOO.dll` name: the `-N` is clearly different.  That might become an issue for _existing_ callers.

The import library that you used (and also renamed) `libFOO.dll.a` -> `libFOO.lib` contains the name change such that you calling the linker with `-lFOO` will find the import library `libFOO.lib` which has the name change to `libFOO-N.dll` -- and thus the _newly created_ callers will do the right thing.
Existing callers will (probably) be looking for the old `libFOO.dll` name.

Or something else might happen.  This is Windows...

Reply:
@ifitchet This is not good, right? Did we come across this when building msys2/uctr64 packages?
```
None INFO    INFO: sysroot: 'C:/Windows/' files: '['winhlp32.exe', 'win.ini', 'twain_32/wiatwain.ds', 'twain_32.dll']'
None INFO WARNING (mpfr,Library/bin/libmpfr-6.dll): $RPATH/libgcc_s_seh-1.dll not found in packages, sysroot(s) nor the
missing_dso_whitelist.
None INFO .. is this binary repackaging?
None INFO WARNING (mpfr,Library/bin/libmpfr-6.dll): $RPATH/libgmp-10.dll not found in packages, sysroot(s) nor the
missing_dso_whitelist.
```

Reply:
It's not great...  However, I believe they are OK.
A vague memory is beginning to form where the Windows executables have two `rpaths` defined in them and something interprets that badly.  Such a discussion might be Slack...

Reply:
Yeah, we discussed this - the .dlls are there; they're actually in Library/ucrt64/bin. This is ok since in Windows, the libraries are found using the PATH, and Library/ucrt64/bin [is in PATH](https://github.com/conda/conda/blob/main/conda/activate.py#L629)

Reply:
This is why I put in the compile test into the test section - to make sure it's still working ok despite overlinking errors being turned off

Reply:
> Existing callers will (probably) be looking for the old libFOO.dll name.

Good catch. I will copy libmpfr-6.dll to mpfr.dll, the old name